nav2_dwb_controller/dwb_plugins/src/kinematic_parameters.cpp

```diff
@@ -140,11 +141,28 @@ void KinematicsHandler::setSpeedLimit(const double & speed_limit)
     kinematics.max_vel_y_ = kinematics.base_max_vel_y_;
     kinematics.max_vel_theta_ = kinematics.base_max_vel_theta_;
   } else {
-    // Speed limit is expressed in % from maximum speed of robot
-    kinematics.max_speed_xy_ = kinematics.base_max_speed_xy_ * speed_limit / 100.0;
-    kinematics.max_vel_x_ = kinematics.base_max_vel_x_ * speed_limit / 100.0;
-    kinematics.max_vel_y_ = kinematics.base_max_vel_y_ * speed_limit / 100.0;
-    kinematics.max_vel_theta_ = kinematics.base_max_vel_theta_ * speed_limit / 100.0;
+    if (percentage) {
+      // Speed limit is expressed in % from maximum speed of robot
+      kinematics.max_speed_xy_ = kinematics.base_max_speed_xy_ * speed_limit / 100.0;
+      kinematics.max_vel_x_ = kinematics.base_max_vel_x_ * speed_limit / 100.0;
+      kinematics.max_vel_y_ = kinematics.base_max_vel_y_ * speed_limit / 100.0;
+      kinematics.max_vel_theta_ = kinematics.base_max_vel_theta_ * speed_limit / 100.0;
+    } else {
+      // Speed limit is expressed in absolute value
+      if (speed_limit < kinematics.base_max_speed_xy_) {
```

Comment:
These 3 `speed_limit < ...` checks are a little fragile, what happens if one update is OK then the second isn't. Then rather than resetting to the base values, they'd continue to use older speed restricted values. Which... might be the best behavior but wouldn't be expected behaviors.

I can understand why you added these checks, but I wonder if we should assume by the time you hit this point in the code that the inputs are valid so that they don't need to be here.

Reply:
These checks are made for the case of incorrect component velocity or even of input data. For example, lets assume we have omni-drive with following `base_` values: `base_max_speed_xy = 5 m/s; base_max_speed_x = 4 m/s; base max_speed_y = 3 m/s`. What will happen if we set `speed_limit = 3.5 m/s`? We can not just set `max_speed_y = 3.5 m/s` because it will more than conceived by the robot manufacturer (and can even damage the robot). Probably, it should be: `max_speed_xy = 3.5` and `max_speed_x = 4 * (3.5/5), max_speed_y = 3 * (3.5/5)`. This at will preserve robot trajectory and will not go beyond the permitted speed. Currently, it has another logic. I will update it in next commit.

> I can understand why you added these checks

Because there we guarantee that speed restriction will be >= 0 m/s, but nobody checks that it less than `base_` value.

Reply:
Update: starting from here changed as described above https://github.com/ros-planning/navigation2/pull/2149/commits/6bd018bee773e8ae042062cde5963646a93cba25#diff-459f50d709535d07d499cd88e84154787ed2f224f5f2beb5f6317b7cb886fbfdL154